openlibrary/templates/type/edition/view.html

```diff
@@ -497,3 +502,57 @@ <h2 class="home-h2"><a href="$page.url()/lists">$_('Lists containing this Book')
       </tbody>
     </table>
 </div>
+
+$def get_identifier(label):
+    $ identifier = list(values[0].value for name, values in edition.get_identifiers().multi_items() if values[0].label == label)
+    $ identifier = identifier[0] if identifier else None
+    $return identifier
+
+$def get_url(label):
+    $ url = list(values[0].url for name, values in edition.get_identifiers().multi_items() if values[0].label == label)
+    $ url = url[0] if url else None
+    $return url
+    
+
+<script type="application/ld+json">
+{
```

Comment:
Constructing the JSON here could be a little error-prone ; we should move this logic to a new function inside the work and edition model (maybe `as_json_ld(work, edition=None)`). Then we can build the json as a python `dict`, and convert it to a JSON string using `json.dumps`. Let me know if you have any questions!

Reply:
Where should I call this function? In `type/edition/view.html` or somewhere else?

Reply:
Yep! Exactly. You should be able to do `work.as_json_ld()` or `edition.as_json_ld()` or `page.as_json_ld()`.